include/path_tracking_pid/details/derivative.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <path_tracking_pid/details/fifo_array.hpp>
+
+namespace path_tracking_pid::details
+{
+/**
+ * @brief discrete time derivative filter
+ */
+class Derivative
+{
+public:
+  /**
+   * @brief Construct a Derivative instance
+   */
+  Derivative();
```

Comment:
Consider removing this. It's not required. The default ctor generated by the compiler does the exact same thing.

Reply:
Done